optimade/models/structures.py

```diff
@@ -751,6 +757,59 @@ class StructureResourceAttributes(EntryResourceAttributes):
         queryable=SupportLevel.MUST,
     )
 
+    @validator("chemical_formula_reduced", "chemical_formula_hill")
+    def check_ordered_formula(cls, v, field):
+        numbers = re.findall(r"(\d+\.|\d+)+", v)
+        elements = re.findall(r"[A-Z][a-z]?", v)
+        expected_elements = sorted(elements)
+
+        if field.name == "chemical_formula_hill":
+            for elem in ("C", "H"):
+                if elem in expected_elements:
+                    _ = expected_elements.pop(expected_elements.index(elem))
+                    expected_elements = [elem] + expected_elements
+
+        if any(elem not in CHEMICAL_SYMBOLS for elem in elements):
+            raise ValueError(
+                f"Cannot use unknown chemical symbols {[elem for elem in elements if elem not in CHEMICAL_SYMBOLS]} in {field.name!r}"
+            )
```

Comment:
I get your concern, but we're already validating e.g. `species_at_sites` to be something in this list. I can't see a use-case where you would want to put anything else in this field...

Reply:
Sure.
I guess it's something we should discuss in plenum at some point, preferrably together with some people more involved with the specification than the Python tools.
For now, since we're already doing this other places, it's fine to leave this in the PR.